trainTestReinforcementAlgorithm.py

```diff
@@ -115,15 +138,39 @@ def saveRL(
     model.save(model_filename)  
 
 
+# %% [markdown]
+# ## Initialize Game Variables
+# %%
+# Set amount of time for training/testing. One step is one action for the snake.
+train_timesteps=1000
+test_timesteps=100
+
+# Set gym environment name
+env_name = 'snake-v0'
+
+# Set board dimensions
+board_height = 10
+board_width = 10
+
+# Set visualization arguments
+visualize_training = False # We don't want to visualize the training process
+visualize_testing = True # Set to true in order to see game moves in pygame. Should be false if run on server.
+visualization_fps = 30
```

Comment:
I think that having these variables out here within the jupyter notebook is a lot less clean than just having them as default values within each function. I think this for a couple reasons:
* Defaults within functions are less likely to be lost over time when we make future code changes
* Defaults within functions makes it clearer for people new to the code base what the intended functionality of everything is.
* Makes running things on the server easier. Both bc we don't need to uncomment these extra cells and bc the defaults are already programmed so we don't have to pass in all the args

Reply:
One potential benefit I saw for setting up defaults in this way for the Jupyter setup was that now we don't run the risk of having mismatching configuration between running/training (i.e. a 10x10 board for training and 5x5 for testing, something I was dumb enough to do while working on this PR) which could be nice.

Additionally, for the third suggestion, the CL arguments still have defaults, so I'm not sure how this setup affects that use case?

I'm totally happy to implement the suggestions and agree with you entirely about your second point, but I'm curious what you think about the above before I change it up again ^^

Reply:
Hmm I think that in 99% of the cases we are going to use a 10x10 board so if those are the defaults I think that the likelihood of a mismatch would be small. In some cases you might imagine wanting to train on a 5x5 but testing on a 10x10 out of curiousity.

Yeah agreed that the CL argument defaults are helpful. My point was that when we run this file from the CL, it run the cell that you created with game parameters even though none of these parameters will be used. This is a minor inconvenience. More importantly, we need to remember to comment out the lines that test and train within the jupyter notebook. Reducing clutter from all the other stuff makes it easier to remember to comment the right things. 

At the end of the day, this is really just a stylistic issue. There's not a real right answer. I obviously prefer to get rid of the game parameters but it is not critical by any means. A lot of my thinking on this was informed by [The Philosophy of Software Design](https://www.amazon.com/Philosophy-Software-Design-John-Ousterhout/dp/1732102201) which really emphasizes the importance of defaults